tests/queries/0_stateless/02302_column_decl_null_before_defaul_value.sql

```diff
@@ -0,0 +1,61 @@
+select 'create table, column +type +NULL';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id INT NULL) ENGINE=MergeTree() ORDER BY tuple();
+DESCRIBE TABLE null_before;
+
+select 'create table, column +type +NOT NULL';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id INT NOT NULL) ENGINE=MergeTree() ORDER BY tuple();
+DESCRIBE TABLE null_before;
+
+select 'create table, column +type +NULL +DEFAULT';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id INT NULL DEFAULT 1) ENGINE=MergeTree() ORDER BY tuple();
+DESCRIBE TABLE null_before;
+
+select 'create table, column +type +NOT NULL +DEFAULT';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id INT NOT NULL DEFAULT 1) ENGINE=MergeTree() ORDER BY tuple();
+DESCRIBE TABLE null_before;
+
+select 'create table, column +type +DEFAULT +NULL';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id INT DEFAULT 1 NULL) ENGINE=MergeTree() ORDER BY tuple();
+DESCRIBE TABLE null_before;
+
+select 'create table, column +type +DEFAULT +NOT NULL';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id INT DEFAULT 1 NOT NULL) ENGINE=MergeTree() ORDER BY tuple();
+DESCRIBE TABLE null_before;
+
+select 'create table, column -type +NULL +DEFAULT';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id NULL DEFAULT 1) ENGINE=MergeTree() ORDER BY tuple();
+DESCRIBE TABLE null_before;
+
+select 'create table, column -type +NOT NULL +DEFAULT';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id NOT NULL DEFAULT 1) ENGINE=MergeTree() ORDER BY tuple();
+DESCRIBE TABLE null_before;
+
+select 'create table, column -type +DEFAULT +NULL';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id DEFAULT 1 NULL) ENGINE=MergeTree() ORDER BY tuple();
+DESCRIBE TABLE null_before;
+
+select 'create table, column -type +DEFAULT +NOT NULL';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id DEFAULT 1 NOT NULL) ENGINE=MergeTree() ORDER BY tuple();
+DESCRIBE TABLE null_before;
+
+select 'alter column, NULL modifier is not allowed';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id INT NOT NULL) ENGINE=MergeTree() ORDER BY tuple();
+ALTER TABLE null_before ALTER COLUMN id TYPE INT NULL; -- { clientError SYNTAX_ERROR }
+
+select 'modify column, NULL modifier is not allowed';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id INT NOT NULL) ENGINE=MergeTree() ORDER BY tuple();
+ALTER TABLE null_before MODIFY COLUMN id NULL DEFAULT 1; -- { serverError UNKNOWN_TYPE }
```

Comment:
I wonder why it's not client error and not `SYNTAX_ERROR`

Reply:
It's a bit unexpected, but I think it's not necessary to fix.